### PR TITLE
Update test runner output with colors and diffs

### PR DESCRIPTION
**Pull request by dylan-conway**
Added [diffz](https://github.com/ziglibs/diffz) for character and line diffs. The function `diffLines()` is the only change to the library.

Object line diff
Screenshot 2023-02-20 at 9 39 31 PM

String character diff
Screenshot 2023-02-20 at 9 40 25 PM

Output changes for other matchers follow this format.
Before:
Screenshot 2023-02-20 at 9 44 09 PM
After:
Screenshot 2023-02-20 at 9 37 10 PM

### CODE REVIEW

File: src/bun.js/bindings/bindings.cpp [hunk 1]
```diff
@@ -97,27 +97,23 @@ static void copyToUWS(WebCore::FetchHeaders* headers, UWSResponse* res)
     auto& internalHeaders = headers->internalHeaders();
 
     for (auto& value : internalHeaders.getSetCookieHeaders()) {
-        res->writeHeader(std::string_view("set-cookie", 10), std::string_view(
-            value.is8Bit() ? reinterpret_cast<const char*>(value.characters8()) : value.utf8().data(), value.length()
-        ));
+        res->writeHeader(std::string_view("set-cookie", 10), std::string_view(value.is8Bit() ? reinterpret_cast<const char*>(value.characters8()) : value.utf8().data(), value.length()));
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.cpp:100]: you have clang-format right

File: src/bun.js/bindings/bindings.cpp [hunk 2]
```diff
@@ -3762,6 +3751,40 @@ void JSC__JSValue__forEachProperty(JSC__JSValue JSValue0, JSC__JSGlobalObject* g
...
+
+    JSC::VM& vm = globalObject->vm();
+    auto scope = DECLARE_CATCH_SCOPE(vm);
+
+    JSC::PropertyNameArray properties(vm, PropertyNameMode::StringsAndSymbols, PrivateSymbolMode::Exclude);
+    JSC::JSObject::getOwnPropertyNames(object, globalObject, properties, DontEnumPropertiesMode::Include);
+
+    Vector<std::pair<String, JSValue>> ordered_properties;
+    for (auto property : properties) {
+        JSValue propertyValue = object->getDirect(vm, property);
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.cpp:3774]: is this correct? we don't want to call getters or prototype values? what happens if you pass it a Response subclass?

File: src/bun.js/bindings/bindings.zig [hunk 3]
```diff
@@ -3638,6 +3647,19 @@ pub const JSValue = enum(JSValueReprInt) {
...
 
+    pub const DiffMethod = enum(u8) {
+        none,
+        character,
+        word,
+        line,
+    };
+
+    pub fn determineDiffMethod(this: JSValue, other: JSValue, global: *JSGlobalObject) DiffMethod {
+        if ((this.isString() and other.isString()) or (this.jsType() == .RegExpObject and other.jsType() == .RegExpObject) or (this.isBuffer(global) and other.isBuffer(global))) return .character;
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/bindings.zig:3658]: we could still make `jsType() == .RegExpObject` a `fn isRegExp(this: JSValue) bool`

File: src/bun.js/bindings/bindings.cpp (hunk 1 above)
Comment (dylan-conway) [reply src/bun.js/bindings/bindings.cpp:100]: yup, these are formatting changes after merging main

File: src/bun.js/test/jest.zig [hunk 4]
```diff
@@ -61,6 +63,161 @@ fn notImplementedProp(
...
+        }
+
+        {
+            var buffered_writer_ = bun.MutableString.BufferedWriter{ .context = &received_buf };
+            var buffered_writer = &buffered_writer_;
+
+            var buf_writer = buffered_writer.writer();
+            const Writer = @TypeOf(buf_writer);
+
+            JSC.ZigConsoleClient.format(
```
Comment (Jarred-Sumner) [inline src/bun.js/test/jest.zig:87]: we probably should make the options to `format` a Struct because its getting a lot of fields

Comment (dylan-conway) [reply src/bun.js/test/jest.zig:87]: 👍

File: src/bun.js/bindings/bindings.zig (hunk 3 above)
Comment (dylan-conway) [reply src/bun.js/bindings/bindings.zig:3658]: 👍

**Comment by Jarred-Sumner** [comment]
thank you @SuperAuguste for the diff library!

**Pull request merged**